### PR TITLE
refactor: only use cache when required

### PR DESCRIPTION
**Pull request by KSXGitHub**

### CODE REVIEW

File: crates/tarball/src/lib.rs [hunk 1]
```diff
@@ -122,9 +121,12 @@ pub struct DownloadTarballToStore<'a> {
 }
 
 impl<'a> DownloadTarballToStore<'a> {
-    /// Execute the subroutine.
-    pub async fn run(self) -> Result<Arc<HashMap<OsString, PathBuf>>, TarballError> {
-        let &DownloadTarballToStore { tarball_cache, package_url, .. } = &self;
+    /// Execute the subroutine with cache.
+    pub async fn with_cache(
```
Comment (anonrig) [inline crates/tarball/src/lib.rs:125]: Nit:

```suggestion
    pub async fn run_with_cache(
```

File: crates/tarball/src/lib.rs [hunk 2]
```diff
@@ -153,15 +155,16 @@ impl<'a> DownloadTarballToStore<'a> {
...
             let mut cache_write = cache_lock.write().await;
             *cache_write = CacheValue::Available(Arc::clone(&cas_paths));
             notify.notify_waiters();
             Ok(cas_paths)
         }
     }
 
-    async fn without_cache(&self) -> Result<Arc<HashMap<OsString, PathBuf>>, TarballError> {
+    /// Execute the subroutine without a cache.
+    pub async fn without_cache(&self) -> Result<HashMap<OsString, PathBuf>, TarballError> {
```
Comment (anonrig) [inline crates/tarball/src/lib.rs:167]: Nit

```suggestion
    pub async fn run_without_cache(&self) -> Result<HashMap<OsString, PathBuf>, TarballError> {
```

**Review by anonrig — APPROVED** [review]

**Review by zkochan — COMMENTED** [review]
Why is it called a tarball cache at all? I had to look into the code because I thought it was about a filesystem cache for tarballs. I'd say it is a tarball_locker. And the function would be `run_with_lock` and `run_without_lock`.

**Comment by KSXGitHub** [comment]
@zkochan `lock` would be even more ambiguous and misleading name than `cache`. Anyway, I have rename the functions to `run_with[out]_mem_cache` and the `tarball_cache` fields to `tarball_mem_cache` (`mem` stands for "in-memory").

File: crates/tarball/src/lib.rs (hunk 1 above)
Comment (KSXGitHub) [reply crates/tarball/src/lib.rs]: > a function should be a verb

BTW, Rust naming convention isn't like that.

**Review by zkochan — APPROVED** [review]

**Pull request merged**